scalanet/src/io/iohk/scalanet/peergroup/dynamictls/DynamicTLSPeerGroup.scala

```diff
@@ -141,6 +142,95 @@ object DynamicTLSPeerGroup {
     }
   }
 
+  sealed abstract case class FramingConfig private (
+      maxFrameLength: Int,
+      lengthFieldOffset: Int,
+      lengthFieldLength: Int,
+      encodingLengthAdjustment: Int,
+      decodingLengthAdjustment: Int,
+      initialBytesToStrip: Int,
+      failFast: Boolean,
+      byteOrder: ByteOrder,
+      lengthIncludesLengthFieldLength: Boolean
+  )
+
+  object FramingConfig {
+    private val possibleLengthFieldLength = Set(1, 2, 3, 4, 8)
+
+    /**
+      *
+      * Configures framing format for all the peers in PeerGroup
+      *
+      *
+      * @param maxFrameLength the maximum length of the frame. If the length of the frame is greater than this value
+      *                       channel with generate DecodingError event.
+      * @param lengthFieldOffset the offset of the length field.
+      * @param lengthFieldLength  the length of the length field. Must be 1, 2, 3, 4 or 8 bytes.
+      * @param encodingLengthAdjustment the compensation value to add to the value of the length field when encoding.
+      * @param decodingLengthAdjustment the compensation value to add to the value of the length field when decoding.
+      * @param initialBytesToStrip the number of first bytes to strip out from the decoded frame. In standard framing setup
+      *                            it should be equal to lengthFieldLength.
+      * @param failFast  If true, a DecodingError event is generated as soon as the decoder notices the length of the frame will
+      *                  exceed maxFrameLength regardless of whether the entire frame has been read. If false,
+      *                  a a DecodingError event is generated after the entire frame that exceeds maxFrameLength has been read..
+      * @param byteOrder the ByteOrder of the length field.
+      * @param lengthIncludesLengthFieldLength if true, the length of the prepended length field is added to the value of the prepended length field.
+      */
+    def buildConfig(
+        maxFrameLength: Int,
+        lengthFieldOffset: Int,
+        lengthFieldLength: Int,
+        encodingLengthAdjustment: Int,
+        decodingLengthAdjustment: Int,
+        initialBytesToStrip: Int,
+        failFast: Boolean = true,
+        byteOrder: ByteOrder = ByteOrder.BIG_ENDIAN,
+        lengthIncludesLengthFieldLength: Boolean = false
+    ): Option[FramingConfig] = {
+      val tooLargeOffset = lengthFieldOffset > maxFrameLength - lengthFieldLength
```

Comment:
So the `maxFrameLength` includes the prefix. Shouldn't there be a validation to check whether `maxFrameLength` can even be expressed with `lengthFieldLength` bytes?

Reply:
from what I know those checks are done when encoding and decoding messages. The only checks I have added here are the checks which netty does inside their constructors i.e it will always be possible to construct netty encoder and decoder from properly constructed `FramingConfig`

Reply:
I mean if I created a config with `maxFrameLength = 65535, lengthFieldLength = 1, lengthFieldOffset = 0`, then you could not capture the maximum length with the available bits in the length field. I don't know if netty constructors would complain, but this condition doesn't capture it.

Reply:
so situation here is that constructors would not complain. The failure would happen when trying to encode message which would violate this `lengthFieldLength` field i.e all would be fine when smaller messages would be sent, and only sending messages larger than 255 would fail

Reply:
Then I suggest adding these to the validations you do :)